registry/storage/driver/testsuites/testsuites.go

```diff
@@ -424,180 +405,190 @@ func (suite *DriverSuite) testContinueStreamAppend(c *check.C, chunkSize int64)
-	c.Assert(err, check.IsNil)
-	c.Assert(writer.Size(), check.Equals, curSize)
+	suite.Require().NoError(err)
+	suite.Require().Equal(curSize, writer.Size())
 
 	nn, err = io.Copy(writer, bytes.NewReader(contentsChunk2))
-	c.Assert(err, check.IsNil)
-	c.Assert(nn, check.Equals, int64(len(contentsChunk2)))
+	suite.Require().NoError(err)
+	suite.Require().Equal(int64(len(contentsChunk2)), nn)
 
 	err = writer.Close()
-	c.Assert(err, check.IsNil)
+	suite.Require().NoError(err)
 
 	curSize = writer.Size()
-	c.Assert(curSize, check.Equals, 2*chunkSize)
+	suite.Require().Equal(2*chunkSize, curSize)
 
 	writer, err = suite.StorageDriver.Writer(suite.ctx, filename, true)
-	c.Assert(err, check.IsNil)
-	c.Assert(writer.Size(), check.Equals, curSize)
+	suite.Require().NoError(err)
+	suite.Require().Equal(curSize, writer.Size())
 
 	nn, err = io.Copy(writer, bytes.NewReader(fullContents[curSize:]))
-	c.Assert(err, check.IsNil)
-	c.Assert(nn, check.Equals, int64(len(fullContents[curSize:])))
+	suite.Require().NoError(err)
+	suite.Require().Equal(int64(len(fullContents[curSize:])), nn)
 
 	err = writer.Commit(context.Background())
-	c.Assert(err, check.IsNil)
+	suite.Require().NoError(err)
 	err = writer.Close()
-	c.Assert(err, check.IsNil)
+	suite.Require().NoError(err)
 
 	received, err := suite.StorageDriver.GetContent(suite.ctx, filename)
-	c.Assert(err, check.IsNil)
-	c.Assert(received, check.DeepEquals, fullContents)
+	suite.Require().NoError(err)
+	suite.Require().Equal(fullContents, received)
 }
 
 // TestReadNonexistentStream tests that reading a stream for a nonexistent path
 // fails.
-func (suite *DriverSuite) TestReadNonexistentStream(c *check.C) {
+func (suite *DriverSuite) TestReadNonexistentStream() {
 	filename := randomPath(32)
 
 	_, err := suite.StorageDriver.Reader(suite.ctx, filename, 0)
-	c.Assert(err, check.NotNil)
-	c.Assert(err, check.FitsTypeOf, storagedriver.PathNotFoundError{})
-	c.Assert(strings.Contains(err.Error(), suite.Name()), check.Equals, true)
+	suite.Require().Error(err)
+	suite.Require().IsType(err, storagedriver.PathNotFoundError{})
+	suite.Require().Contains(err.Error(), suite.Name())
 
 	_, err = suite.StorageDriver.Reader(suite.ctx, filename, 64)
-	c.Assert(err, check.NotNil)
-	c.Assert(err, check.FitsTypeOf, storagedriver.PathNotFoundError{})
-	c.Assert(strings.Contains(err.Error(), suite.Name()), check.Equals, true)
+	suite.Require().Error(err)
+	suite.Require().IsType(err, storagedriver.PathNotFoundError{})
+	suite.Require().Contains(err.Error(), suite.Name())
 }
 
 // TestWriteZeroByteStreamThenAppend tests if zero byte file handling works for append to a Stream
-func (suite *DriverSuite) TestWriteZeroByteStreamThenAppend(c *check.C) {
+func (suite *DriverSuite) TestWriteZeroByteStreamThenAppend() {
+	if suite.StorageDriver.Name() == "s3aws" {
+		// See https://github.com/distribution/distribution/pull/4185#discussion_r1419721968
+		suite.T().Skip("S3 multipart uploads require at least 1 chunk (>0B)")
+	}
```

Comment:
```suggestion
```
This test asserts that an implementation upholds a particular aspect of the storage driver interface contract. The s3 storage driver is failing to uphold its side of the contract, therefore the s3 storage driver contains a bug that needs fixing. "S3 multipart uploads require at least 1 chunk" is the root cause of the bug, not an excuse for the driver to not adhere to the contract.

Reply:
Corrected in https://github.com/distribution/distribution/pull/4185/commits/e77c81c50b1fba53e4c6d32d641e0d55246f6664